### DISCUSSION TITLE
Implement  Spatial Query Raycast sorting like with Raycaster/RayHits component

### DISCUSSION THREAD

**Discussion by jacobrolph98** [avianphysics/avian#262] [Ideas]
Using the Raycast/hit components, RayHits implements iter_sorted() to sort results by time_of_impact. There is no such feature when using Spatial Query raycasting, as it returns Vec<RayHitData>. Cannot use vec.sort as RayHitData doesn't implement Ord, etc

Please let me know if I'm missing something.

**Comment by Jondolf** [comment]
You can sort the vector with `sort_by`:

```rust
hits.sort_by(|a, b| a.time_of_impact.partial_cmp(&b.time_of_impact).unwrap())
```

`RayHitData` currently doesn't implement `Ord` for the same reason `Vec<f32>` doesn't implement `Ord`; floats inherently aren't totally ordered due to precision issues. Instead, `Ord` would have to be like a `PartialOrd` implementation and use `partial_cmp` like the code above, which probably wouldn't be semantically correct. For usability reasons it could be nice though
